tests/framework/pkg/clientbase/ops.go

```diff
@@ -207,6 +209,18 @@ func (a *APIOperations) DoCreate(schemaType string, createObj interface{}, respO
 		return a.DoResourceDelete(schemaType, &obj)
 	})
 
+	a.Session.RegisterWaitFunc(func() error {
```

Comment:
we should not wait by default in the client, let the developer decide what needs to be synchronous or asynchronous.  Additionally the session should only be used for cleanups.

Reply:
It's not waiting by default, I'm registering it's wait function for when we cleanup we can safely cleanup all the resources created.